server/s3_get_object_action.cc

```diff
@@ -475,6 +497,25 @@ void S3GetObjectAction::send_data_to_client() {
     LOG_PERF("get_object_send_data_ms", request_id.c_str(), mss);
     s3_stats_timing("get_object_send_data", mss);
 
+    if (S3Option::get_instance()->get_s3_read_md5_check_enabled()) {
+      std::string checksum_calculated;
+      if (object_metadata->get_part_one_size() == 0) {
+        checksum_calculated = motr_reader->get_content_md5();
+      } else {
+        checksum_calculated = motr_reader->awsetag.finalize();
+      }
+      std::string checksum_read = object_metadata->get_md5();
+      s3_log(S3_LOG_DEBUG, request_id,
+             "checksum calculated: %s, checksum read %s",
+             checksum_calculated.c_str(), checksum_read.c_str());
+      if (checksum_calculated != checksum_read) {
+	s3_iem(LOG_ERR, S3_IEM_CHECKSUM_MISMATCH,
```

Comment:
On my screen, it looks like the indentation here is not correct.  Is this a hard tab vs soft tab issue maybe?

Reply:
It's a well-known GitHub issue EOS-11827.

S3server uses spaces. Fixed here and in the rest of the patch, all tabs were found with `git diff cortx-1.0 | grep -P '^\+\t'`.